File: mobile/sistem_pos/lib/models/product.dart
class Product {
  const Product({
    required this.id,
    required this.categoryId,
    required this.name,
    required this.price,
    this.description,
    this.sku,
    this.imageUrl,
    this.isActive = true,
  });

  final int id;
  final int categoryId;
  final String name;
  final num price;
  final String? description;
  final String? sku;
  final String? imageUrl;
  final bool isActive;

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['id'] as int,
        categoryId: json['category_id'] as int,
        name: json['name'] as String,
        price: json['price'] as num,
        description: json['description'] as String?,
        sku: json['sku'] as String?,
        imageUrl: json['image_url'] as String?,
        isActive: json['is_active'] as bool? ?? true,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'category_id': categoryId,
        'name': name,
        'price': price,
        'description': description,
        'sku': sku,
        'image_url': imageUrl,
        'is_active': isActive,
      };
}

File: backend/app/schemas/product.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field, field_serializer


class ProductBase(BaseModel):
    category_id: int
    name: str = Field(min_length=1, max_length=100)
    description: str | None = None
    sku: str | None = Field(default=None, max_length=50)
    price: Decimal = Field(gt=0, max_digits=12, decimal_places=2)
    image_url: str | None = None
    is_active: bool = True


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    category_id: int | None = None
    name: str | None = Field(default=None, min_length=1, max_length=100)
    description: str | None = None
    sku: str | None = Field(default=None, max_length=50)
    price: Decimal | None = Field(default=None, gt=0, max_digits=12, decimal_places=2)
    image_url: str | None = None
    is_active: bool | None = None


class ProductOut(ProductBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime
    updated_at: datetime

    @field_serializer("price")
    def serialize_price(self, value: Decimal) -> float:
        return float(value)


class ProductList(BaseModel):
    items: list[ProductOut]
    total: int
    page: int
    page_size: int